plugins/woocommerce/tests/e2e-pw/tests/admin-analytics/analytics-data.spec.js

```diff
@@ -0,0 +1,322 @@
+		await expect( page.getByRole( 'menuitem', { name: 'Taxes $0.00 No change from Previous year:' } ) ).toBeVisible();
+		await expect( page.getByRole( 'menuitem', { name: 'Shipping $0.00 No change from Previous year:' } ) ).toBeVisible();
+		await expect( page.getByRole( 'menuitem', { name: 'Total sales $0.00 No change from Previous year:' } ) ).toBeVisible();
+	} );
+
+	test( 'set custom date range on revenue report', async( { page } ) => {
+		await page.goto( '/wp-admin/admin.php?page=wc-admin&path=%2Fanalytics%2Frevenue' );
+
+		// assert that current month is shown and that values are for that
+		await expect( page.getByText( 'Month to date' ) ).toBeVisible();
+		await expect( page.getByRole( 'menuitem', { name: 'Gross sales $1,229.30 No change from Previous year:' } ) ).toBeVisible();
+		await expect( page.getByRole( 'menuitem', { name: 'Returns $0.00 No change from Previous year:' } ) ).toBeVisible();
+		await expect( page.getByRole( 'menuitem', { name: 'Coupons $0.00 No change from Previous year:' } ) ).toBeVisible();
+		await expect( page.getByRole( 'menuitem', { name: 'Net sales $1,229.30 No change from Previous year:' } ) ).toBeVisible();
+		await expect( page.getByRole( 'menuitem', { name: 'Taxes $0.00 No change from Previous year:' } ) ).toBeVisible();
+		await expect( page.getByRole( 'menuitem', { name: 'Shipping $0.00 No change from Previous year:' } ) ).toBeVisible();
+		await expect( page.getByRole( 'menuitem', { name: 'Total sales $1,229.30 No change from Previous year:' } ) ).toBeVisible();
+
+		// click the date filter and change to custom date range (should be no sales/orders)
+		await page.getByRole( 'button', { name: 'Month to date' } ).click();
+		await page.getByText( 'Custom', { exact: true } ).click();
+		await page.getByPlaceholder( 'mm/dd/yyyy' ).first().fill( '01/01/2022' );
+		await page.getByPlaceholder( 'mm/dd/yyyy' ).last().fill( '01/30/2022' );
+		await page.getByRole( 'button', { name: 'Update' } ).click();
+
+		// assert values updated
+		await expect( page.getByRole( 'button', { name: 'Custom (Jan 1 - 30, 2022) vs. Previous year (Jan 1 - 30, 2021)'} ) ).toBeVisible();
+		await expect( page.getByRole( 'menuitem', { name: 'Gross sales $0.00 No change from Previous year:' } ) ).toBeVisible();
+		await expect( page.getByRole( 'menuitem', { name: 'Returns $0.00 No change from Previous year:' } ) ).toBeVisible();
+		await expect( page.getByRole( 'menuitem', { name: 'Coupons $0.00 No change from Previous year:' } ) ).toBeVisible();
+		await expect( page.getByRole( 'menuitem', { name: 'Net sales $0.00 No change from Previous year:' } ) ).toBeVisible();
+		await expect( page.getByRole( 'menuitem', { name: 'Taxes $0.00 No change from Previous year:' } ) ).toBeVisible();
+		await expect( page.getByRole( 'menuitem', { name: 'Shipping $0.00 No change from Previous year:' } ) ).toBeVisible();
+		await expect( page.getByRole( 'menuitem', { name: 'Total sales $0.00 No change from Previous year:' } ) ).toBeVisible();
```

Comment:
I actually don't think this really helps that much.  We're just abstracting the `getByRole()` and `isVisible()` parts of the assertion.  Those aren't likely to change, but I could see the code change in the future where one of these can't be a `getByRole()` and we have to use a different locator.  Then we'd have to re-write this all again.

Yes, there is some duplication, but I think you can also look at the code and see exactly what we're asserting/expecting here instead of having to go look up what the `detailsToCheck` function does.

Reply:
I think this is one of those few cases where repetition makes the test code more readable. I'll go ahead and resolve this convo as I think it's ok to be explicit in repetitive UI assertions such as this.